Submodule src/golang.org/x/tools contains modified content
diff --git a/src/golang.org/x/tools/cmd/present/static/styles.css b/src/golang.org/x/tools/cmd/present/static/styles.css
index d877682b..7c4d0f37 100644
--- a/src/golang.org/x/tools/cmd/present/static/styles.css
+++ b/src/golang.org/x/tools/cmd/present/static/styles.css
@@ -265,6 +265,13 @@
   letter-spacing: -1px;
 }
 
+.slideNum {
+    font-size: 0.75em;
+    position: absolute;
+    bottom: 0px;
+    right: 4px;
+}
+
 b {
   font-weight: 600;
 }
@@ -382,7 +389,8 @@ code {
   font-size: 95%;
   font-family: 'Droid Sans Mono', 'Courier New', monospace;
 
-  color: black;
+/* color: black; */
+  color: blue;
 }
 
 article > .image,
diff --git a/src/golang.org/x/tools/cmd/present/templates/action.tmpl b/src/golang.org/x/tools/cmd/present/templates/action.tmpl
index 8bde042c..e4d10e2c 100644
--- a/src/golang.org/x/tools/cmd/present/templates/action.tmpl
+++ b/src/golang.org/x/tools/cmd/present/templates/action.tmpl
@@ -4,7 +4,7 @@ It determines how the formatting actions are rendered.
 */}
 
 {{define "section"}}
-  <h{{len .Number}} id="TOC_{{.FormattedNumber}}">{{.FormattedNumber}} {{.Title}}</h{{len .Number}}>
+  <h{{len .Number}} id="TOC_{{.FormattedNumber}}">{{.FormattedNumber}} {{style .Title}}</h{{len .Number}}>
   {{range .Elem}}{{elem $.Template .}}{{end}}
 {{end}}
 
diff --git a/src/golang.org/x/tools/cmd/present/templates/slides.tmpl b/src/golang.org/x/tools/cmd/present/templates/slides.tmpl
index 878440b6..af789d71 100644
--- a/src/golang.org/x/tools/cmd/present/templates/slides.tmpl
+++ b/src/golang.org/x/tools/cmd/present/templates/slides.tmpl
@@ -53,15 +53,19 @@
         {{end}}
       </article>
 
+  {{$numSlides := len .Sections}}
   {{range $i, $s := .Sections}}
   <!-- start of slide {{$s.Number}} -->
       <article>
       {{if $s.Elem}}
-        <h3>{{$s.Title}}</h3>
+        <h2>{{style $s.Title}}</h2>
         {{range $s.Elem}}{{elem $.Template .}}{{end}}
       {{else}}
-        <h2>{{$s.Title}}</h2>
+        <h2>{{style $s.Title}}</h2>
       {{end}}
+      <span class="slideNum">
+	{{$i}}/{{$numSlides}}
+      </span>
       </article>
   <!-- end of slide {{$i}} -->
   {{end}}{{/* of Slide block */}}
